API/RAT.m

```diff
@@ -1,4 +1,58 @@
 function [project,result] = RAT(project,controls)
+% Runs RAT calculation for the given project and controls inputs.
+%
+% Parameters
+% ----------
+% project : projectClass
+%    An instance of the ``projectClass``.
+% controls : controlsClass
+%    An instance of the ``controlsClass``.   
+%
+% Returns
+% -------
+% project : projectClass
+%    An instance of the ``projectClass`` with updated fit values from the calculation.
+% result : struct
+%    The result of the calculation such as simulated reflectivities, SLD profiles etc.
+%
+% Notes
+% -----
+% For a non-Bayesian procedure, the result struct has the following fields below where nDomains, 
+% nContrasts and nParams are the the number of domains, contrasts and params respectively  
+%  
+% ================== ============================ ===============
+% Field              Type                         Description
+% ================== ============================ ===============
+% reflectivity       [nContrasts x 1] cell        calculated reflectivity values
+% simulation         [nContrasts x 1] cell        simulated reflectivity values
+% shiftedData        [nContrasts x 1] cell        Data shifted using given scale factor
+% backgrounds        [nContrasts x 1] cell        background values for each contrast
+% resolutions        [nContrasts x 1] cell        resolution values for each contrast 
+% layerSlds          [nContrasts x nDomains] cell layers parameter values for each contrast
+% sldProfiles        [nContrasts x nDomains] cell SLD profiles
+% resampledLayers    [nContrasts x nDomains] cell resampled layer values for each contrast 
+%                                                 if resample is true otherwise an array of zeros
+% calculationResults [1 x 1] struct               fields are 'chiValues' and 'sumChi'  
+% contrastParams     [1 x 1] struct               fields are 'scalefactors', 'bulkIn', 'bulkOut', 
+%                                                 'subRoughs', and 'resample'
+% fitParams          [1 x nParams] double         fitted parameter values
+% fitNames           [nParams x 1] cell           name of fitted parameters
+% ================== ============================ ===============
+%
+% For a Bayesian procedure, the result struct will also contain the following field in 
+% addition to the ones above  
+%  
+% =================== ==================== ===============
+% Field               Type                 Description
+% =================== ==================== ===============
+% predictionIntervals struct               fields are 'reflectivity', 'sld', and 'sampleChi'
```

Comment:
"Mean, 65% and 95% confidence intervals, and chi squared goodness of fit values for the reflectivity and SLD."